### PR TITLE
ytdl_hook.lua: add include script-opt of URLs to try with ytdl first

### PR DESCRIPTION
**Pull request by guidocella**
By the default mpv tries opening URLs with ffmpeg first, and users who don't configure try_ytdl_first get a slower startup for youtube URLs, on top of yt-dlp already being slow.

Fix this by adding a script-opt of URL pattern to try with ytdl first. Youtube and twitch are included by default.

Compared to the alternative of trying ytdl first by default and excluding URLs with a media extension, this only works for the sites explicitly included in the new option, but doesn't have false positives on media URLs without an extension.

Alternative to #14963

### CODE REVIEW

**Comment by kasper93** [comment]
I'm ok with this change, it's not like some urls would work with ytdl anyway.

File: DOCS/man/options.rst [hunk 1]
```diff
@@ -917,6 +917,13 @@ Program Behavior
 
         See more lua patterns here: https://www.lua.org/manual/5.1/manual.html#5.4.1
 
+    ``include=<URL1|URL2|...``
+        A ``|``-separated list of URL patterns which mpv should try to parse with
+        youtube-dl first when ``try_ytdl_first`` is ``no``. The patterns are
+        matched in the same way as ``exclude``.
+
+        Default: ``youtu%.?be|twitch%.tv``
```
Comment (kasper93) [inline DOCS/man/options.rst:925]: This `youtu%.?be` is little bit too ambiguous for my taste.

File: DOCS/man/options.rst [hunk 2]
```diff
@@ -899,9 +899,9 @@ Program Behavior
         on whether most of your URLs need youtube-dl parsing.
 
     ``exclude=<URL1|URL2|...``
-        A ``|``-separated list of URL patterns which mpv should not use with
-        youtube-dl. The patterns are matched after the ``http(s)://`` part of
-        the URL.
+        A ``|``-separated list of URL patterns which mpv should not try parsing
+        with youtube-dl first when ``try_ytdl_first`` is ``yes``. The patterns
+        are matched after the ``http(s)://`` part of the URL.
```
Comment (kasper93) [inline DOCS/man/options.rst:904]: Why this is updated? The logic didn't change? This should exclude use of ytdl for those url, regardless of `try_ytdl_first`

Comment (guidocella) [reply DOCS/man/options.rst:904]: Seems I misunderstood how it works.

**Review by kasper93 — APPROVED** [review]

**Pull request merged**